controllers/clusterversion_controller.go

```diff
@@ -356,3 +369,49 @@ func (c *ClusterVersionReconciler) getOperatorConfig() (*corev1.ConfigMap, error
 	}
 	return cm, nil
 }
+
+func (c *ClusterVersionReconciler) ensureConsolePlugin() error {
+	nginxConf := console.GetNginxConfiguration()
+
+	consoleDeployment := console.GetDeployment(c.OperatorNamespace)
+	err := c.Client.Get(context.TODO(), types.NamespacedName{
+		Name:      consoleDeployment.Name,
+		Namespace: consoleDeployment.Namespace,
+	}, consoleDeployment)
+	if err != nil {
+		return err
+	}
+
+	consoleConfigMap := console.GetNginxConfConfigMap(c.OperatorNamespace)
+	_, err = controllerutil.CreateOrUpdate(context.TODO(), c.Client, consoleConfigMap, func() error {
+		if consoleConfigMapData := consoleConfigMap.Data["nginx.conf"]; consoleConfigMapData != nginxConf {
+			c.log.Info(fmt.Sprintf("Set the ConfigMap odf-client-console-nginx-conf data as '%s'", nginxConf))
+			consoleConfigMap.Data["nginx.conf"] = nginxConf
+		}
+		return controllerutil.SetControllerReference(consoleDeployment, consoleConfigMap, c.Scheme)
+	})
+
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+
+	// Create/Update ODF console Service
+	consoleService := console.GetService(c.ConsolePort, c.OperatorNamespace)
+	_, err = controllerutil.CreateOrUpdate(context.TODO(), c.Client, consoleService, func() error {
+		return controllerutil.SetControllerReference(consoleDeployment, consoleService, c.Scheme)
+	})
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+
+	// Create/Update ODF console ConsolePlugin
+	consolePlugin := console.GetConsolePluginCR(c.ConsolePort, c.OperatorNamespace)
+	_, err = controllerutil.CreateOrUpdate(context.TODO(), c.Client, consolePlugin, func() error {
+		return nil
```

Comment:
you might need to do update inside the function to replace the spec, please check